src/VaultV2.sol

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: UNLICENSED
```

Comment:
not sure what this VaultV2 is....didnt review

Reply:
V2 is the update as per the stretch goals to take in any two ERC20 tokens - potentially with different decimals, as mentioned in the description